core/src/main/java/kafka/metrics/ClientMetricsConfigs.java

```diff
@@ -80,6 +82,11 @@ public class ClientMetricsConfigs {
     public static final String CLIENT_SOURCE_ADDRESS = "client_source_address";
     public static final String CLIENT_SOURCE_PORT = "client_source_port";
 
+    // Empty string in client-metrics resource configs indicates that all the metrics are subscribed.
+    public static final String ALL_SUBSCRIBED_METRICS_CONFIG = "\"\"";
```

Comment:
This is what the user provides. Why is `ALL_SUBSCRIBED_METRICS` not the same as this?

Reply:
The KIP mentions that the response in subscriptions should be just empty string i.e. `""`. But to create `client-metrics` resource the ConfigDef parses `""` as no data hence to specify empty string through `kafka-configs.sh` we need to pass them as empty string enclosed in string i.e. `"\"\""`

Reply:
It would be possible to change the KIP to support a list containing just `*` to mean all metrics. The KIP today says a list containing just `""` means all metrics. That's a pretty simple change.